lib/Transforms/Scalar/UnifyAssumes.cc

```diff
@@ -135,6 +181,34 @@ bool UnifyAssumesPass::runOnFunction(Function &F) {
   return false;
 }
 
+void UnifyAssumesPass::processAssertInst(CallInst &CI, AllocaInst &flag) {
+  BasicBlock *bb = CI.getParent();
+  Module *M = bb->getParent()->getParent();
+  assert(bb);
+  assert(M);
+
+  auto &SBI = getAnalysis<seahorn::SeaBuiltinsInfoWrapperPass>().getSBI();
+  llvm::CallSite CS(&CI);
+  IRBuilder<> B(bb);
+  B.SetInsertPoint(&CI);
+  auto ante = B.CreateLoad(&flag);
+  // negate condition if verifier.assert.not seen
+  bool isNot = SBI.getSeaBuiltinOp(CI) == seahorn::SeaBuiltinsOp::ASSERT_NOT;
+  Value *cond = CS.getArgument(0);
+  if (isNot) {
+    cond = B.CreateNot(cond);
+  }
+  // transform verifier.assert(cond) into sea.assert.if(flag, cond)
+  Function *assertIfFn =
+      SBI.mkSeaBuiltinFn(seahorn::SeaBuiltinsOp::ASSERT_IF, *M);
+  CallInst *NewCI = CallInst::Create(assertIfFn, {ante, cond});
+  NewCI->setCallingConv(assertIfFn->getCallingConv());
+  if (!CI.use_empty())
+    CI.replaceAllUsesWith(NewCI);
+  ReplaceInstWithInst(&CI, NewCI);
```

Comment:
check whether this moved debug location from old to new instruction

Reply:
Yes.
/// Replace the instruction specified by From with the instruction specified by
/// To. Copies DebugLoc from BI to I, if I doesn't already have a DebugLoc.

Reply:
I bet `ReplaceInstWithInst` also takes care of updating all the users, so lines `185--186` are not necessary.